pkg/kube/inject/inject.go

```diff
@@ -763,10 +764,11 @@ func IntoObject(sidecarTemplate string, valuesConfig string, meshconfig *meshcon
 	}
 
 	metadata.Annotations[annotation.SidecarStatus.Name] = status
+	if metadata.Labels == nil {
```

Comment:
Does this need to be a label as opposed to an annotation?

Reply:
Your choice.

I thought it would be cool to be able to say `kubectl get pods --all-namespaces --selector istio.io/rev=canary`.  Where the revision hangs in the metadata should not affect my planned control plane finding implementation for istioctl.